File: red-scare/java/Edge.java
public class Edge {
    Node from;
    Node to;

    public Edge(Node from, Node to) {
        this.from = from;
        this.to = to;
    }

    public boolean adjecentToRed(){
        return from.isRed || to.isRed;
    }

    public boolean adjecentToSource(){
        return from.isSource || to.isSource;
    }

    public boolean adjecentToSink(){
        return from.isSink || to.isSink;
    }
}

File: red-scare/jon/rs/Edge.kt
package rs

data class Edge(
        val label: String,
        val from: Node,
        val to: Node
) {
    val adjacentToRed: Boolean = from.isRed || to.isRed
    val adjacentToSource: Boolean = from.isSource || to.isSource
    val adjacentToSink: Boolean = from.isSink || to.isSink

    constructor(): this(
            "",
            Node(false, "", false, false),
            Node(false, "", false, false)
    )
}
